modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -273,80 +302,73 @@ package object engine {
       case SetSkyBackground(sb)    => log("Engine: Setting sky background") *> setSkyBackground(sb)
       case SetCloudCover(cc)       => log("Engine: Setting cloud cover") *> setCloudCover(cc)
       case Poll                    => log("Engine: Polling current state")
-      case Exit                    => log("Engine: Bye") *> close(q)
       case GetState(f)             => getState(f)
       case Log(msg)                => log(msg)
     }
 
-    def handleSystemEvent(se: SystemEvent): Handle[Unit] = se match {
+    def handleSystemEvent(se: SystemEvent): HandleP[Unit] = se match {
       case Completed(id, i, r)     => log("Engine: Action completed") *> complete(id, i, r)
       case PartialResult(id, i, r) => log("Engine: Partial result") *> partialResult(id,i, r)
-      case Failed(id, i, e)        => log("Engine: Action failed") *> fail(q)(id)(i, e)
+      case Failed(id, i, e)        => log("Engine: Action failed") *> fail(id)(i, e)
       case Busy(id)                => log("Engine: Resources needed this sequence are busy")
-      case Executed(id)            => log("Engine: Execution completed") *> next(q)(id)
-      case Executing(id)           => log("Engine: Executing") *> execute(q)(id)
-      case Finished(id)            => log("Engine: Finished") *> switch(q)(id)(SequenceState.Completed)
+      case Executed(id)            => log("Engine: Execution completed") *> next(id)
+      case Executing(id)           => log("Engine: Executing") *> execute(id)
+      case Finished(id)            => log("Engine: Finished") *> switch(id)(SequenceState.Completed)
     }
 
     (ev match {
         case EventUser(ue)   => handleUserEvent(ue)
         case EventSystem(se) => handleSystemEvent(se)
-      }) *> get
+    })*> get
   }
 
   // Kudos to @tpolecat
   /** Traverse a process with a stateful computation. */
-  private def mapEvalState[F[_]: Monad: Catchable, A, S, B](
-    fs: Process[F, A], s: S, f: A => StateT[F, S, B]
-  ): Process[F, B] = {
-    def go(fs: Process[F, A], s: S): Process[F, B] =
-      Process.eval(fs.unconsOption).flatMap {
-        case None         => Process.halt
-        case Some((h, t)) => Process.eval(f(h).run(s)).flatMap {
-          case (s, a) => Process.emit(a) ++ go(t, s)
+  private def mapEvalState[A, S, B](
```

Comment:
Very nice. As an extended challenge you could abstract the types for `Option` and `Task` 😄